Assets/Scripts/EnemyController.cs

```diff
@@ -16,130 +18,120 @@ public class EnemyController : MonoBehaviour
     public int maxSpeed;
     public int shieldRegen;
     public float weight;
-    public string movementType;
+    public MovementType movementType;
     public Vector3 movementTarget;
     public List<Transform> path; // hierarchy objects should be ordered
     public int currentPathTarget = 0;
     public float distanceTolerance;
-
+    public float speed = 0.05f;
 
     private Rigidbody2D enemyRigidBody;
+
+    // START Unity magic functions
     void OnDeath()
     {
-        duration = Time.time-spawnTime;
-        Debug.Log("A " + type + " enemy died after " + duration+ " seconds!");
+        duration = Time.time - spawnTime;
+        Debug.Log(string.Format("A {0} enemy died after {1} seconds!", type, duration));
```

Comment:
This is to show a better way of doing string creation when you need parameterization: string.Format() is a cool little standard utility in C# that makes it easier to deal with strings when you have parameters. Instead of having to do a bunch of pluses and open/close quotations for string snippets, you can just type out the whole string you want in one go, and replace the bits you want as parameters with {#}, where # is the index of the parameter passed in at the end.

This also comes in handy if you want to use the same parameter twice: for example, "{0} {0} {0}" would output the first parameter (in this case 'type') 3 times in a row.